multicluster/service-mirror/cluster_watcher.go

```diff
@@ -655,6 +707,75 @@ func (rcsw *RemoteClusterServiceWatcher) Start(ctx context.Context) error {
 			},
 		},
 	)
+	// Add an event handler for Endpoints object to see whether we need to
+	// update mirrored headless services. We only care if an Endpoints is
+	// updated and if it belongs
+	// to a headless service.
+	rcsw.remoteAPIClient.Endpoint().Informer().AddEventHandler(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
```

Comment:
Looks like AddFunc and UpdateFunc handlers are similar enough to be extracted into a separate function?